builtin/providers/azure/resource_azure_virtual_network_test.go

```diff
@@ -0,0 +1,295 @@
+						"azure_virtual_network.foo", &network),
+					testAccCheckAzureVirtualNetworkAttributes(&network),
+					resource.TestCheckResourceAttr(
+						"azure_virtual_network.foo", "name", "terraform-vnet"),
+					resource.TestCheckResourceAttr(
+						"azure_virtual_network.foo", "location", "West US"),
+					resource.TestCheckResourceAttr(
+						"azure_virtual_network.foo", "address_space.0", "10.1.2.0/24"),
+					resource.TestCheckResourceAttr(
+						"azure_virtual_network.foo", "subnet.33778499.name", "subnet1"),
+					resource.TestCheckResourceAttr(
+						"azure_virtual_network.foo", "subnet.33778499.address_prefix", "10.1.2.0/25"),
+					resource.TestCheckResourceAttr(
+						"azure_virtual_network.foo", "subnet.33778499.security_group", "terraform-security-group1"),
+				),
+			},
+
+			resource.TestStep{
+				Config: testAccAzureVirtualNetwork_update,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAzureVirtualNetworkExists(
+						"azure_virtual_network.foo", &network),
+					testAccCheckAzureVirtualNetworkAttributes(&network),
+					resource.TestCheckResourceAttr(
+						"azure_virtual_network.foo", "name", "terraform-vnet"),
+					resource.TestCheckResourceAttr(
+						"azure_virtual_network.foo", "location", "West US"),
+					resource.TestCheckResourceAttr(
+						"azure_virtual_network.foo", "address_space.0", "10.1.3.0/24"),
+					resource.TestCheckResourceAttr(
+						"azure_virtual_network.foo", "subnet.514595123.name", "subnet1"),
+					resource.TestCheckResourceAttr(
+						"azure_virtual_network.foo", "subnet.514595123.address_prefix", "10.1.3.128/25"),
+					resource.TestCheckResourceAttr(
+						"azure_virtual_network.foo", "subnet.514595123.security_group", "terraform-security-group2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAzureVirtualNetworkExists(
+	n string,
+	network *virtualnetwork.VirtualNetworkSite) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Virtual Network ID is set")
+		}
+
+		mc := testAccProvider.Meta().(*Client).mgmtClient
+		nc, err := virtualnetwork.NewClient(mc).GetVirtualNetworkConfiguration()
+		if err != nil {
+			return err
+		}
+
+		for _, n := range nc.Configuration.VirtualNetworkSites {
+			if n.Name == rs.Primary.ID {
+				*network = n
+
+				return nil
+			}
+		}
+
+		return fmt.Errorf("Virtual Network not found")
+	}
+}
+
+func testAccCheckAzureVirtualNetworkAttributes(
+	network *virtualnetwork.VirtualNetworkSite) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		if network.Name != "terraform-vnet" {
+			return fmt.Errorf("Bad name: %s", network.Name)
+		}
+
+		if network.Location != "West US" {
+			return fmt.Errorf("Bad location: %s", network.Location)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckAzureVirtualNetworkDestroy(s *terraform.State) error {
```

Comment:
Indeed a good catch! I think I might have used this (wrong) approach in the CloudStack provider as well. Will walk through both providers and update the tests where applicable...